### ISSUE TITLE
missing glyphs

### ISSUE THREAD

**Issue by javadr** [SajadKazemi/Parsian#1]
Firstly, I would like to express my gratitude for your efforts in developing a high-quality font for Persian users within the Tech community, specifically for applications like LaTeX. 

Secondly, I have conducted a thorough comparison between your font and two other fonts, namely [Amiri](https://www.amirifont.org/) and [Scheherazade](https://software.sil.org/scheherazade/). In order to provide you with a clear illustration of my observations, I recommend running the following Minimum Working Example code:

```latex
\documentclass{article}
\usepackage[landscape]{geometry}
\usepackage[uthmani]{quran}
\usepackage{pgffor}
\usepackage{multicol}
\usepackage{fullpage}
\usepackage[computeautoilg=on]{xepersian}
\settextfont{Parsian}
\defpersianfont\fonta{Amiri}
\defpersianfont\fontb{Scheherazade}

\begin{document}
    \foreach \x in {1,...,604} {
        \begin{multicols}{3}
            \quranpage[\x]
            \newcolumn
            {\fonta \quranpage[\x]}
            \newcolumn
            {\fontb \quranpage[\x]}
        \end{multicols}
        \newpage
    }
    
\end{document}```

Upon examination, I noticed that your font lacks several crucial Arabic glyphs, especially when it comes to typesetting the Holy Quran. These missing glyphs are as follows:

1. (U+06D6) 'ARABIC SMALL HIGH LIGATURE SAD WITH LAM WITH ALEF MAKSURA'
2. (U+06D7) 'ARABIC SMALL HIGH LIGATURE QAF WITH LAM WITH ALEF MAKSURA'
3. (U+06D8) 'ARABIC SMALL HIGH MEEM ISOLATED FORM'
4. (U+06D9) 'ARABIC SMALL LOW ALEF'
5. (U+06DA) 'ARABIC SMALL HIGH JEEM'
6. (U+06DB) 'ARABIC SMALL HIGH THREE DOTS'
7. (U+06DC) 'ARABIC SMALL HIGH SEEN'
8. (U+06DF) 'ARABIC SMALL HIGH ROUNDED ZERO'
9. (U+06E0) 'ARABIC SMALL HIGH UPRIGHT RECTANGULAR ZERO'
10. (U+06E2) 'ARABIC SMALL HIGH MEEM INITIAL FORM'
11. (U+06E3) 'ARABIC SMALL LOW SEEN'
12. (U+06E5) 'ARABIC SMALL WAW'
13. (U+06E6) 'ARABIC SMALL YEH'
14. (U+06E7) 'ARABIC SMALL HIGH YEH'
15. (U+06E8) 'ARABIC SMALL HIGH NOON'
16. (U+06E9) 'ARABIC PLACE OF SAJDAH'
17. (U+06EA) 'ARABIC EMPTY CENTRE LOW STOP'
18. (U+06EB) 'ARABIC EMPTY CENTRE HIGH STOP'
19. (U+06EC) 'ARABIC ROUNDED HIGH STOP WITH FILLED CENTRE'
20. (U+06ED) 'ARABIC SMALL LOW MEEM'

These glyphs are vital for accurately representing and typesetting the Holy Quran.

**Comment by SajadKazemi** [comment]
Thanks.

To be honest, I'm a little bit confused about **Parsian** being compared to masterpieces like **Amiri** or **Scheherazade**. **Amiri** and **Scheherazade** are so sophisticated that I personally have to spend a day or two examining their extensive OpenType features. 

Before talking about the Holy Quran, let me ask this: Is there any issue with ordinary Arabic texts like what they type in their everyday online conversations, websites or scientific textbooks? If no, then I think **Parsian** is good to go.

The whole idea behind **Parsian** was to be a fine, standard font for Latin & Persian _language_ (not necessarily Persian people) that looks good in mathematical and scientific contexts.

I Personally have no practical knowledge of _any_ of the glyphs you mentioned above; and as I see it, they are exclusive to the Holy Quran, not even regular Arabic texts. So unless the issue is about Persian language or at least typical Arabic texts which are identical to Persian in sense of letters and marks, with all due respect, I'm afraid there's nothing I can do about it now.

BUT, if you think you can tell me more about the nature of the marks that are exclusive to the Holy Quran, like their usage, placement, conditions under which marks would appear, etc, I like to hear about it.

**Comment by javadr** [comment]
I do not agree with your idea. We have lots of Persian texts which include Arabic transcripts and even some parts of the Holy Quran. From another perspective, if you want to design a decent font for Persian users, your font must include a wide range of glyphs declared by the Unicode consortium. 

By the way, I am unable to make you do that, but I strongly suggest that you add these glyphs to make your font widely acceptable. Take a look at the "Tahoma" font created by Microsoft during the early age of Windows; it also includes all of the aforementioned glyphs. 

Good luck!

**Comment by SajadKazemi** [comment]
Quranic marks are added to the fonts. Both **Parsian** and **Parsian Sans** now contain glyphs mentioned in the issue.

Thanks for your suggestion, and since you're the author of `quran` package, I appreciate it if you run some additional tests on fonts to see if there's any other missing Quranic marks.

**Comment by javadr** [comment]
I appreciate your effort and consideration. 
That's great to hear that we now have all the glyphs. 
At a glance, it seems that all the added glyphs are positioned a bit far from the left character and closer to the right one. I believe it should be the other way around. 
Please take a look at the pictures below to see what I mean.

![image](https://github.com/SajadKazemi/Parsian/assets/1178221/8d7553a8-f152-43d3-b3f5-8383e5a2fdf2)
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/d5f12992-de3f-49ed-a885-c538497081a4)
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/7baf9421-6a31-4998-9582-d1710e1c7c97)
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/d11ec17d-ea36-4973-b6b2-570abe3583b7)
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/c4e18977-fffc-48af-8130-ace0d2a0069b)
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/04b76cca-476d-42fc-a901-ae3a7e71f4da)
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/a81cb2ca-f47c-41af-b988-d03ddc5c3deb)
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/fe65948d-1b59-46d0-877e-ec8a71257077)
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/6e40b2c6-91ef-438a-ba4d-4b61aab9e268)

There is also another issue with the word بالاخره
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/cf2e4d74-b089-4ade-84e8-4106f5c39700)
should be like below
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/3fe356e9-4444-454f-9906-43c22d3c7d8b)

**Comment by javadr** [comment]
Could you make the following symbol smaller?
![image](https://github.com/SajadKazemi/Parsian/assets/1178221/cb170560-69af-4b9e-962e-6bf47ee8c6e5)

**Comment by SajadKazemi** [comment]
New fonts uploaded.

A note about "بالاخره": The font has nothing to do with that. It would be better if you review the actual plain text that's in the package. Because turning "ا+ء" into "أ" is something one would normally see in fonts for special needs like **Amiri Quran**. No standard font with Arabic support would have this kind of rule. Even **Tahoma** and **Arial** don't have this feature. 
This substitution can be added to the font, but it will turn **Parsian** fonts into a _special font_ like **Amiri Quran** and may turn normal literature into strange words.

Other issues have been fixed. Please test the fonts and let me know if it works well with texts from the Holy Quran.
